developer/src/kmc-kmn/src/compiler/compiler.ts

```diff
@@ -213,6 +213,22 @@ export class KmnCompiler implements UnicodeSetParser {
     return result;
   }
 
+  /**
+   * By default, when a `Uint8Array` is created from an `ArrayBuffer` (e.g.
+   * `Module.HEAP8.buffer`), it is a dynamic view into that buffer. This module
+   * buffer can be dynamically reallocated at any time, which can happen when
+   * allocating memory in WASM code (so the change will  look _really_ weird in
+   * a stack trace). Thus, to ensure we don't trip over ourselves, we need to
+   * copy the buffer. Fortunately, creating a `Uint8Array` from a `Uint8Array`
+   * copies the data, and is pretty quick.
+   * @param offset    Offset into the WASM memory space, in bytes
+   * @param size      Size of the buffer to copy, in bytes
+   * @returns         A _copy_ of the data in a new Uint8Array
+   */
+  private copyWasmBuffer(offset: number, size: number): Uint8Array {
+    return new Uint8Array(new Uint8Array(Module.HEAP8.buffer, offset, size));
```

Comment:
Do we not need to `delete[]` the intermediate `Uint8Array`?

Reply:
It's Javascript, there ain't no `delete` :grin:

Reply:
Oh, right, this one's TS as opposed to the other PRs in the group, which are C++-based.

Reply:
bikeshed: i might have named it `dupWasmBuffer` or `cloneWasmBuffer` or something. `copy` to me sounds like strcpy

Reply:
`copy` to me sounded like `memcpy` :-P

Reply:
`cpy` that.

Reply:
if memcpy took `NULL` as a first argument yeah, meaning "surprise me!" :D